services/terraformer/templates/hetzner-lb.tpl

```diff
@@ -35,15 +35,17 @@ resource "hcloud_firewall" "firewall" {
     ]
   }
 
+  {{range $role := index .Metadata "roles"}}
   rule {
     direction = "in"
-    protocol  = "tcp"
-    port      = "6443"
+    protocol  = "{{ $role.Protocol }}"
+    port      = "{{ $role.Port }}"
     source_ips = [
```

Comment:
Can we limit source IPs to the CIDR block of Wireguard VPN? I think @MarioUhrik suggested that on the last grooming.

Reply:
I think Matus has perfectly nailed it as it is, as far as I could see.

> Can we limit source IPs to the CIDR block of Wireguard VPN?

This is the LB machine. It needs to open these ports to the entire internet. What you're suggesting is nonsense.

The flow is:
- LB nodes open 6443 to the entire Internet
- cluster nodes don't open 6443 at all
- LB nodes forward 6443 to the cluster nodes via their wireguard IP, so only the wireguard firewall rule needs to be present for the cluster nodes

Also:
- In case there is no LB, then cluster nodes open 6443 to the Internet in the same way the LB otherwise would

Please let me know if my understanding is correct @Despire @MiroslavRepka @jaskeerat789

Reply:
>This is the LB machine

My bad, should have finished my coffee first :smile:.

I see the flow the same way you described it, therefore, please disregard my comment

Reply:
Exactly as @MarioUhrik described.